### PR TITLE
Update ManagedVM naming to flex

### PR DESCRIPTION
**Pull request by patflynn**
This change removes 'Managed VM' as a brand and concept for our App
Engine integration.  Where things are clearly Flex only names are
updated accordingly. Otherwise the more generic 'AppEngine' is used.

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ public void actionPerformed(ActionEvent e) {
       }
     });
     userSpecifiedArtifactFileSelector.addBrowseFolderListener(
-        GctBundle.message("appengine.managedvm.config.user.specified.artifact.title"),
+        GctBundle.message("appengine.flex.config.user.specified.artifact.title"),
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java]: This class is renamed have 'AppEngine', but is using a property named 'flex'. Isn't this class Flex specific?

Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java]: Some of these classes have flex specific properties or do flex specific things, but conceivably we might unify that in teh future. If we don't we'll break out standard and flex configurations and rename.

**Comment by etanshaul** [comment]
lgtm

**Comment by patflynn** [comment]
@chanseokoh @meltsufin you guys will want to sync this soon so as to limit merge conflicts.

**Pull request merged**